zulipterminal/helper.py

```diff
@@ -629,3 +633,51 @@ def display_error_if_present(response: Dict[str, Any], controller: Any
                              ) -> None:
     if response['result'] == 'error' and hasattr(controller, 'view'):
         controller.view.set_footer_text(response['msg'], 3)
+
+
+@asynch
+def open_media(controller: Any, media_link: str, message_id: int) -> None:
+    """
+    Downloads media in /tmp/zulip-term/{message_id} directory and uses native
+    viewer for the output.
+    """
+    img_name = media_link.split('/')[-1]
+    target_dir = os.path.join(gettempdir(), 'zulip-term', str(message_id))
```

Comment:
Never write to known paths in the temporary directory. This leads to security vulnerabilities. For example, an attacker with access to a different user account on the system can create a symlink or hardlink at the known path to trick you into overwriting a file of their choice with your permissions. Use `NamedTemporaryFile`, `TemporaryDirectory`, `mkstemp`, or `mkdtemp`, which do the extra work needed to prevent these attacks.